src/radical/entk/appman/wfprocessor.py

```diff
@@ -249,328 +287,235 @@ def _enqueue(self, local_prof):
+                                                 logger=self._logger,
+                                                 reporter=self._report)
+
+                        # Found the pipeline and processed it -- no more
+                        # iterations needed for the current task
+                        break
 
-                except Exception, ex:
-                    self._logger.exception(
-                        'Unable to receive message from completed queue: %s' % ex)
-                    raise
+                # Appease pika cos it thinks the connection is dead
+                now = time.time()
+                if now - last >= self._rmq_ping_interval:
+                    mq_connection.process_data_events()
+                    last = now
 
             self._logger.info('Terminated dequeue thread')
             mq_connection.close()
-
-            local_prof.prof('terminating dequeue-thread', uid=self._uid)
+            self._prof.prof('terminating dequeue-thread', uid=self._uid)
 
         except KeyboardInterrupt:
 
-            self._logger.exception('Execution interrupted by user (you probably hit Ctrl+C), ' +
-                               'trying to exit gracefully...')
+            self._logger.exception('Execution interrupted by user (you \
+                                    probably hit Ctrl+C), trying to exit \
+                                    gracefully...')
 
             mq_connection.close()
-
             raise KeyboardInterrupt
 
-        except Exception, ex:
-            self._logger.exception('Error in dequeue-thread: %s' % ex)
-
+        except Exception:
+            self._logger.exception('Error in dequeue-thread')
             try:
                 mq_connection.close()
-            except:
-                self._logger.warning('mq_connection not created')
+            except Exception as ex:
+                self._logger.warning('mq_connection not created, %s' % ex)
 
-            raise EnTKError(ex)
+            raise
 
-    def _wfp(self):
-        """
-        **Purpose**: This is the function executed in the wfp process. The function is used to simply create
-        and spawn two threads: enqueue, dequeue. The enqueue thread pushes ready tasks to the queues in the pending_q slow
-        list whereas the dequeue thread pulls completed tasks from the queues in the completed_q. This function is also
-        responsible for the termination of these threads and hence blocking.
-        """
+    def _execute_post_exec(self, stage):
 
         try:
 
-            local_prof = ru.Profiler(
-                name='radical.entk.%s' % self._uid + '-proc', path=self._path)
-
-            local_prof.prof('wfp process started', uid=self._uid)
-
-            self._logger.info('WFprocessor started')
+            self._logger.info('Executing post-exec for stage %s' % stage.uid)
+            self._prof.prof('Adap: executing post-exec', uid=self._uid)
 
```

Comment:
I would suggest event names like `post_exec_start` and `post_exec_stop`.

Reply:
Done